lib/server.js

```diff
@@ -41,27 +41,39 @@ var Server = function(server, path, services, wsdl, options) {
   if (path[path.length - 1] !== '/')
     path += '/';
   wsdl.onReady(function(err) {
-    var listeners = server.listeners('request').slice();
-
-    server.removeAllListeners('request');
-    server.addListener('request', function(req, res) {
-      if (typeof self.authorizeConnection === 'function') {
-        if (!self.authorizeConnection(req)) {
-          res.end();
-          return;
+    if(typeof server.route === 'function' && typeof server.use === 'function'){
```

Comment:
`instanceof` wont work in this case. This will work 4x onwards as `use` and `route` are most commonly used express methods. 

Current check gives us an another advantage - we have `express` as dev dependency only :)